### DISCUSSION TITLE
Mapping query to model

### DISCUSSION THREAD

**Discussion by syntext** [prisma/prisma#21845] [Q&A] [answered]
### Question

Is it feasible to directly map SQL queries to model objects at present? The concept here is to define a model (or a database view) within the Prisma database schema. Rather than linking to a table or existing view, the model would correspond to a specific, prewritten SQL query. This could potentially be contained in a separate SQL file. Such an approach would offer the flexibility to craft custom model objects without depending on the physical views present in the database.

I'm curious to know if this capability is available by default, or if I could implement this in an extension function. This function could, for example, map a SQL query within a Common Table Expression (CTE) and treat it akin to a database view.

Anyone has some ideas or can point me in some direction?

**Comment by ludralph** [comment, marked as answer]
Hi @syntext 👋 

Thank you for raising this question.

Currently, Prisma does not support the direct mapping of SQL queries to model objects. There are open GitHub issues discussing the need for such features, including the ability to map raw SQL queries to views without the need for database-side view creation and support, and the addition of SQL expressions in Prisma queries. 

In the first [issue](https://github.com/prisma/prisma/issues/21287), the user suggests the possibility of mapping a raw SQL query to a view, which can be sent from the client without the need to create and support views on the database side. The user proposes two ways to map it: an additional construct can be added directly to the schema, something like @@raw("some-query"), or in the client's configuration code or through the extensions mechanism.

In the second [issue](https://github.com/prisma/prisma/issues/18543), the user suggests adding support for SQL expressions in Prisma queries, which would allow for SQL side modifications of retrieved data without client-side parsing.

Can you please add your use case to the listed issues and also leave a 👍  to help our engineers prioritize it?

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
